py/util.py

```diff
@@ -370,17 +370,14 @@ def load_gdf(geotype: str, basin: int, high_resolution: bool) -> gpd.GeoDataFram
     # Open the shapefile for the basin
     if geotype == "catchments":
         directory = CATCHMENTS_DIR
-        shapefile = f"{directory}/cat_pfaf_{basin}_MERIT_Hydro_v07_Basins_v01.shp"
+        gis_file = f"{directory}/cat_pfaf_{basin}_MERIT_Hydro_v07_Basins_v01.shp"
     elif geotype == "rivers":
-        shapefile = f"{RIVERS_DIR}/riv_pfaf_{basin}_MERIT_Hydro_v07_Basins_v01.shp"
+        gis_file = "https://pub-5f26e013d22e454ea079891d13f905f1.r2.dev/global_flowlines.fgb"
 
-    if not os.path.isfile(shapefile):
-        raise Exception(f"Could not find the file: {shapefile}")
+    if VERBOSE: print(f"Reading geodata in {gis_file}")
+    gdf = gpd.read_file(gis_file, bbox=bounds)
```

Comment:
oops i thought i properly tested this but cached data from a pickle file was being used

re-running things after commenting that out, i'm seeing: 
```
Reading geodata in https://pub-5f26e013d22e454ea079891d13f905f1.r2.dev/global_flowlines.fgb
Traceback (most recent call last):
  File "/home/tngzng/delineator/subbasins.py", line 735, in <module>
    _run_from_terminal()
  File "/home/tngzng/delineator/subbasins.py", line 722, in _run_from_terminal
    delineate(args.input_csv, args.output_prefix)
  File "/home/tngzng/delineator/subbasins.py", line 132, in delineate
    rivers_gdf = load_gdf("rivers", megabasin, True, bounds)
  File "/home/tngzng/delineator/py/util.py", line 380, in load_gdf
    gdf = gpd.read_file(gis_file, bbox=bounds)
  File "/home/tngzng/delineator/venv/lib/python3.10/site-packages/geopandas/io/file.py", line 288, in _read_file
    with urllib.request.urlopen(filename) as response:
  File "/usr/lib/python3.10/urllib/request.py", line 216, in urlopen
    return opener.open(url, data, timeout)
  File "/usr/lib/python3.10/urllib/request.py", line 525, in open
    response = meth(req, response)
  File "/usr/lib/python3.10/urllib/request.py", line 634, in http_response
    response = self.parent.error(
  File "/usr/lib/python3.10/urllib/request.py", line 563, in error
    return self._call_chain(*args)
  File "/usr/lib/python3.10/urllib/request.py", line 496, in _call_chain
    result = func(*args)
  File "/usr/lib/python3.10/urllib/request.py", line 643, in http_error_default
    raise HTTPError(req.full_url, code, msg, hdrs, fp)
urllib.error.HTTPError: HTTP Error 403: Forbidden
```

Reply:
i wonder why fiona has no problem reading the public url, but geopandas does? 🤔